### ISSUE TITLE
[receiver/elasticsearch] Flaky test TestElasticsearchIntegration

### ISSUE THREAD

**Issue by dmitryax** [open-telemetry/opentelemetry-collector-contrib#19755]
Seen in https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/4439992459/jobs/7793226189?pr=19164

```
2023/03/16 17:44:26 Waiting for container id e363a8665428 image: 7c4d8e75-d160-446d-be1d-01df82d17b03:966a30ce-c60b-41b5-9b95-2b7f23a18d73
=== CONT  TestElasticsearchIntegration/Running_elasticsearch_7.16.3
    integration_test.go:131: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/elasticsearchreceiver/integration_test.go:131
        	Error:      	Received unexpected error:
        	            	number of resources doesn't match expected: 4, actual: 0
        	Test:       	TestElasticsearchIntegration/Running_elasticsearch_7.16.3
2023/03/16 17:44:48 Container is ready id: e363a8665428 image: 7c4d8e75-d160-446d-be1d-01df82d17b03:966a30ce-c60b-41b5-9b95-2b7f23a18d73
2023/03/16 17:44:48 Starting container id: e363a8665428 image: 7c4d8e75-d160-446d-be1d-01df82d17b03:966a30ce-c60b-41b5-9b95-2b7f23a18d73
2023/03/16 17:44:48 Waiting for container id e363a8665428 image: 7c4d8e75-d160-446d-be1d-01df82d17b03:966a30ce-c60b-41b5-9b95-2b7f23a18d73
2023/03/16 17:44:48 Container is ready id: e363a8665428 image: 7c4d8e75-d160-446d-be1d-01df82d17b03:966a30ce-c60b-41b5-9b95-2b7f23a18d73
--- FAIL: TestElasticsearchIntegration (0.00s)
    --- FAIL: TestElasticsearchIntegration/Running_elasticsearch_7.16.3 (37.01s)
    --- PASS: TestElasticsearchIntegration/Running_elasticsearch_7.9 (63.79s)
FAIL
coverage: 84.1% of statements
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/elasticsearchreceiver	63.846s
```

**Comment by djaglowski** [comment]
@junhuangli, do you think this could be related to changes made in #19431?

**Comment by junhuangli** [comment]
@djaglowski I hit the same issue last night too(had to trigger a rerun using dummy change). But it has been working fine at least 5-7 times during my check in. Could be a intermittent issue?

**Comment by junhuangli** [comment]
@dmitryax I can see the test passed in the original pull request. Is this still an issue?

**Comment by djaglowski** [comment]
It seems to be intermittent

**Comment by atoulme** [comment]
Occurred again here: https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/4476625651/jobs/7867191314?pr=20232

**Comment by atoulme** [comment]
Sightly different one:
```
=== CONT  TestElasticsearchIntegration/Running_elasticsearch_7.9.3
    integration_test.go:140: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/elasticsearchreceiver/integration_test.go:140
        	Error:      	Received unexpected error:
        	            	resource "map[elasticsearch.cluster.name:docker-cluster elasticsearch.node.version:7.9.3]": scope "otelcol/elasticsearchreceiver": number of metrics doesn't match expected: 57, actual: 54
        	Test:       	TestElasticsearchIntegration/Running_elasticsearch_7.9.3
--- FAIL: TestElasticsearchIntegration (0.00s)
    --- FAIL: TestElasticsearchIntegration/Running_elasticsearch_7.16.3 (52.60s)
    --- FAIL: TestElasticsearchIntegration/Running_elasticsearch_7.9.3 (82.73s)
    --- PASS: TestElasticsearchIntegration/Running_elasticsearch_7.0.0 (91.29s)
```

It is possible that one causes the other.

**Comment by djaglowski** [comment]
Here again: https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/4788238852/jobs/8514586849?pr=21116#step:5:140

**Comment by djaglowski** [comment]
Possibly resolved by #21118. Closing for now but will reopen if issues continue.

**Comment by TylerHelmuth** [comment]
Another example https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/4953726829/jobs/8861516228

**Comment by djaglowski** [comment]
Closed by #22022
